src/clr-angular/data/datagrid/providers/column-order-model.service.ts

```diff
@@ -47,40 +47,20 @@ export class ColumnOrderModelService {
     console.log(dropData);
   }
 
-  private findVisibleColumnModelHasMet(
-    filterCondition: (_model: ColumnOrderModelService) => boolean,
-    reduceCondition: (_reducedModel: ColumnOrderModelService, _currentModel: ColumnOrderModelService) => boolean
-  ): ColumnOrderModelService {
-    const filteredVisibleColumnModels = this.columnOrderCoordinatorService.orderModels.filter(
-      model => !model.isHidden && filterCondition(model)
-    );
-
-    if (filteredVisibleColumnModels.length === 0) {
-      return;
-    }
-
-    return filteredVisibleColumnModels.reduce((reducedModel, currentModel) => {
-      if (reduceCondition(reducedModel, currentModel)) {
-        return currentModel;
-      }
-      return reducedModel;
-    });
+  private findAdjacentVisibleModel(prev = false): ColumnOrderModelService {
+    const filteredVisibleColumnModels = this.columnOrderCoordinatorService.orderModels
+      .filter(model => (!model.isHidden && prev ? model.flexOrder < this.flexOrder : model.flexOrder > this.flexOrder))
```

Comment:
`&&` takes precedence over the ternary operator `? :`, as far as I know. So this doesn't do what you want at all, you need to:
```typescript
!model.isHidden && (prev ? model.flexOrder < this.flexOrder : model.flexOrder > this.flexOrder)
```

Reply:
Fixed :p